provider/clusterrosa/hcp/resource.go

```diff
@@ -578,6 +583,21 @@ func (r *ClusterRosaHcpResource) Create(ctx context.Context, request resource.Cr
 		}
 	}
 
+	if common.HasValue(state.WaitForStdComputeNodesComplete) &&
+		state.WaitForStdComputeNodesComplete.ValueBool() {
+		tflog.Info(ctx, "Waiting for standard compute nodes to get ready")
+		object, err = r.ClusterWait.WaitForStdComputeNodesToBeReady(ctx, object.ID(), rosa.DefaultWaitTimeoutInMinutes)
+		if err != nil {
+			response.Diagnostics.AddError(
+				"Waiting for cluster creation finished with error",
+				fmt.Sprintf("Waiting for cluster creation finished with the error %v", err),
+			)
+			if object == nil {
+				return
+			}
```

Comment:
Is it possible that err is returned and object is not nil?

Reply:
There are instances where even though it errors it expects to populate the object, similarly to error state in cluster waiter